### DISCUSSION TITLE
problem with Bigsur 11.3.1

### DISCUSSION THREAD

**Discussion by Emmanuel78960** [hydrogen-music/hydrogen#1265] [Hydrogen on OS X]
Hi I cannot use my Hydrogen application. Is it a problem with Bigger OS ? I could download it but it doesn't want to open. Apple said there is compatibility problem. Many thanks to help me to solve the.

**Comment by theGreatWhiteShark** [comment]
Hey @Emmanuel78960

Which version of Hydrogen are you using? In both [1.0.2](https://github.com/hydrogen-music/hydrogen/releases/tag/1.0.2) and [1.1.0-beta](https://github.com/hydrogen-music/hydrogen/releases/tag/1.1.0-beta1) this should already be fixed.

**Reply by cme** [reply]
Can you also tell us whether you're on Intel processor or Apple Silicon? And provide a screenshot or transcription of the error you get, please?

As we can't sign binaries (it involves giving a big chunk of cash to Apple) you'll have to right-click the application and hit 'Open' the first time you start it rather than just double-clicking.

**Reply by cme** [reply]
I've just downloaded both releases, and the behaviour seems different on first installation from what I remember.

On the first attempt to open, I get:

<img width="269" alt="image" src="https://user-images.githubusercontent.com/768466/117204176-e1117380-ade7-11eb-852d-7ba17315f4d7.png">

...with no option to just open it.

Hitting "Cancel" on this gets the dialog I *do* remember:

<img width="269" alt="image" src="https://user-images.githubusercontent.com/768466/117204372-23d34b80-ade8-11eb-9e6e-553b130963f9.png">

This might be due to the recent OS update, and might be related to us not signing the binaries at all (in which case we should be able to get round it by adding ad-hoc signing to them. I'll try that in Appveyor and see what happens...)

**Comment by Emmanuel78960** [comment]
Hi thank you for your answer.
I’m using Intel processor on my MacBook Pro
Here are screen copy.
I’ve tried just to use right button to open Hydrogen; but no success, it’s the same message.

Thank for your kindly help.

Emmanuel

De : Colin McEwan ***@***.***>
Répondre à : hydrogen-music/hydrogen ***@***.***>
Date : mercredi 5 mai 2021 à 22:24
À : hydrogen-music/hydrogen ***@***.***>
Cc : Emmanuel78960 ***@***.***>, Mention ***@***.***>
Objet : Re: [hydrogen-music/hydrogen] problem with Bigsur 11.3.1 (#1265)


I've just downloaded both releases, and the behaviour seems different on first installation from what I remember.

On the first attempt to open, I get:

[image]<https://user-images.githubusercontent.com/768466/117204176-e1117380-ade7-11eb-852d-7ba17315f4d7.png>

...with no option to just open it.

Hitting "Cancel" on this gets the dialog I do remember:

[image]<https://user-images.githubusercontent.com/768466/117204372-23d34b80-ade8-11eb-9e6e-553b130963f9.png>

This might be due to the recent OS update, and might be related to us not signing the binaries at all (in which case we should be able to get round it by adding ad-hoc signing to them. I'll try that in Appveyor and see what happens...)

**Comment by Rolf527** [comment]
Opening with the right mouse in MacOS Big Sur 11.3.1 only works if you do this from the Application folder itself. So right click Applications in the dock and choose Open "Applications". Then right click the Hydrogen app. Hope this helps, otherwise please let me know.

**Comment by Emmanuel78960** [comment]
I’m sorry that didn’t work with your method. It’s always the same message

De : Rolf527 ***@***.***>
Répondre à : hydrogen-music/hydrogen ***@***.***>
Date : jeudi 6 mai 2021 à 09:08
À : hydrogen-music/hydrogen ***@***.***>
Cc : Emmanuel78960 ***@***.***>, Mention ***@***.***>
Objet : Re: [hydrogen-music/hydrogen] problem with Bigsur 11.3.1 (#1265)


Opening with the right mouse in MacOS Big Sur 11.3.1 only works if you do this from the Application folder itself. So right click Applications in the dock and choose Open "Applications". Then right click the Hydrogen app. Hope this helps, otherwise please let me know.

**Comment by Rolf527** [comment]
Make sure you open the app in the folder with the right mouse, so you get this message first

![RightClick1](https://user-images.githubusercontent.com/83751246/117260602-8ad81b00-ae4f-11eb-9fc9-4768a154e42c.jpg)

Then choose Open also with right mouse. Then you will get this message.

![RightClick2](https://user-images.githubusercontent.com/83751246/117260712-a4796280-ae4f-11eb-99a5-9e1f9ccf02c5.jpg)

Just tried it with both Hydrogen 1.02 and Hydrogen 1.1.0 beta1.

Have an iMac Pro (intel processor) with Big Sur 11.3.1

Remaining problem:
The startup time of the new Hydrogen versions is currently much longer than with Hydrogen 0.9.7. It is about 15 seconds. Not a real problem but worth mentioning it.

**Reply by theGreatWhiteShark** [reply]
@Rolf527 could we use those images to describe the opening process in the wiki?

@cme How about we extend https://github.com/hydrogen-music/hydrogen/wiki/Building-Hydrogen-from-Source-(macOS) or create a parent page in the wiki to handle all the MacOS stuff. There we could also go into the particularities of the individual versions of the OS.

**Comment by Emmanuel78960** [comment]
GOOOOOOD !!!! it’s working !!
Many thanks
Emmanuel

De : Rolf527 ***@***.***>
Répondre à : hydrogen-music/hydrogen ***@***.***>
Date : jeudi 6 mai 2021 à 09:46
À : hydrogen-music/hydrogen ***@***.***>
Cc : Emmanuel78960 ***@***.***>, Mention ***@***.***>
Objet : Re: [hydrogen-music/hydrogen] problem with Bigsur 11.3.1 (#1265)


Make sure you open the app in the folder with the right mouse, so you get this message first

[RightClick1]<https://user-images.githubusercontent.com/83751246/117260602-8ad81b00-ae4f-11eb-9fc9-4768a154e42c.jpg>

Then choose Open also with right mouse. Then you will ge this message.

[RightClick2]<https://user-images.githubusercontent.com/83751246/117260712-a4796280-ae4f-11eb-99a5-9e1f9ccf02c5.jpg>

Just tried it with both Hydrogen 1.02 and Hydrogen 1.1.0 beta1.

Have an iMac Pro (intel processor) with Big Sur 11.3.1

**Reply by Emmanuel78960** [reply]
Hi after holidays I tried to used Hydrogen and there is a problem with.
I didn't see the arrow when the song runs and after 5 bars I got troubles (there square dance ou shake ...  sorry it's difficult to explain it in english)

**Reply by theGreatWhiteShark** [reply]
> I didn't see the arrow when the song runs and after 5 bars I got troubles (there square dance ou shake ... sorry it's difficult to explain it in english)

Hey @Emmanuel78960 ,

Can you open an issue for this problem and provide some further infos, like which version of Hydrogen, audio driver, steps to reproduce, in there?

**Reply by cme** [reply]
It might help to post a video of the behaviour if describing it is difficult. You can use Quicktime Player (File -> New screen recording) to capture video.

**Comment by Rolf527** [comment]
The startup time of Hydrogen on my iMac reduced from 15 sec to under 1 sec if the Audio System is changed in Preferences from auto to core audio.

![Startup](https://user-images.githubusercontent.com/83751246/117275848-c201f880-ae5e-11eb-8400-3b5c389ff363.jpg)

**Reply by theGreatWhiteShark** [reply]
> The startup time of Hydrogen on my iMac reduced from 15 sec to under 1 sec

Ouch. This sounds like a bug.

If the `Auto` driver is selected (per default), Hydrogen tries to use a number of different driver in the following order: 
1. `JACK`
2. `ALSA`
3. `CoreAudio`
4. ...

At least in Linux the `JACK` driver tries to connect to or start the JACK server (if not running yet) twice before deciding to hand over to the next driver, which - as `ALSA` is probably not available in macOs - is the `CoreAudio` one.

@cme Do you think `JACK` is more important than `CoreAudio` for MacOS user? If not, we could use an alternative order for this platform (as we already do for Windows).

**Reply by cme** [reply]
Agreed, defaulting to CoreAudio on macOS seems the right thing to do, though I wonder if that means we should make the *default* sound system CoreAudio rather than "Auto", rather than making "Auto" prefer CoreAudio -- CoreAudio should always be available so "Auto" would always pick it, rather than detecting an alternate audio system the user might want to use.

This also reminds me I need to get round to fixing JACK on Apple Silicon...

**Reply by theGreatWhiteShark** [reply]
Hmmm. I think making "Auto" prefer CoreAudio has two advantages. 1. By making "Auto" default option on macOS we provide a more consistent UX across all platforms. 2. We advertise the "Auto" option in the manual as beginner option for all users who have no idea about audio drivers/systems at all. More advanced users should have no problem selecting the JACK driver on their own.

**Comment by Rolf527** [comment]
Of course, no problem.

Added also a comment (with image) on the startup time (15 sec on my iMac Pro). This could be reduced to under 1 sec by changing the audio system from auto to coreaudio in Hydrogen preferences.

Am very happy now with the new Hydrogen versions for MacOS!

Best regards/Rolf

**Comment by Rolf527** [comment]
I don't have the JACK driver installed on my iMac. I do have Airfoil installed, perhaps that gives the trouble here. Am always using CoreAudio as a Logic Pro user.

**Comment by Rolf527** [comment]
Forgot to mention that version 0.9.7 did not have that problem. Perhaps that gives a clue.
